bluemira/equilibria/analysis.py

```diff
@@ -0,0 +1,1479 @@
+            f"I_p = {eq.profiles.I_p / 1e6:.2f} MA \n"
+        )
+        eq.profiles.plot(ax=ax2)
+        ax2.set_title("Profiles")
+        plt.suptitle(eq.label)
+        if show:
+            plt.show()
+        return ax1, ax2
+
+    def plot_compare_separatrix(
+        self,
+        title=None,
+        ax=None,
+        show=True,  # noqa: FBT002
+    ):
+        """
+        Plot separatrices of input equilibria and reference equilibria.
+        N.B. Plots LCFS if a fixed boundary is used.
+
+        Parameters
+        ----------
+        title:
+            Title to be added at top of figure
+        ax:
+            Matplotlib Axes objects set by user
+        show:
+            Whether or not to display the plot
+        equilibrium_name:
+            Name used in plot label
+
+        Returns
+        -------
+        ax:
+            Matplotlib Axes object
+
+        """
+        eq = self._get_input()
+        ref = self._get_reference()
+        if ax is None:
+            _, ax = plt.subplots()
+
+        eq_fs = eq.get_separatrix() if isinstance(eq, Equilibrium) else eq.get_LCFS()
+
+        if isinstance(ref, Equilibrium):
+            ref_eq_fs = ref.get_separatrix()
+        else:
+            ref_eq_fs = ref.get_LCFS()
+
+        if isinstance(eq_fs, list):
+            ax.plot(eq_fs[0].x, eq_fs[0].z, color="red", label=eq.label)
+            ax.plot(eq_fs[1].x, eq_fs[1].z, color="red")
+        else:
+            ax.plot(eq_fs.x, eq_fs.z, color="red", label=eq.label)
+
+        if isinstance(ref_eq_fs, list):
+            ax.plot(
+                ref_eq_fs[0].x,
+                ref_eq_fs[0].z,
+                color="blue",
+                linestyle="--",
+                label=ref.label,
+            )
+            ax.plot(ref_eq_fs[1].x, ref_eq_fs[1].z, color="blue", linestyle="--")
+        else:
+            ax.plot(
+                ref_eq_fs.x,
+                ref_eq_fs.z,
+                color="blue",
+                linestyle="--",
+                label=ref.label,
+            )
+
+        ax.legend(loc="best")
+        ax.set_xlabel("$x$ [m]")
+        ax.set_ylabel("$z$ [m]")
+        ax.set_aspect("equal")
+
+        if title is not None:
+            plt.suptitle(title)
+        if show:
+            plt.show()
+        return ax
+
+    def plot_compare_psi(
+        self,
+        diag_ops: EqDiagnosticOptions | None = None,
+        ax=None,
+    ):
+        """
+        Plot Psi comparision.
+
+        Parameters
+        ----------
+        diag_ops:
+            Diagnostic plotting options
+        ax:
+            List of Matplotlib Axes objects set by user
+
+        Raises
+        ------
+        BluemiraError
+            if no refernce equilibrium is provided
+
+        Returns
+        -------
+        :
+            plotting class
+
+        """
+        eq = self._get_input()
+        ref = self._get_reference()
+        if diag_ops is not None:
+            self.diag_ops = diag_ops
```

Comment:
```suggestion
        diag_ops = diag_ops or self.diag_ops
```

I don't think you should set the state here

Reply:
Also you would need to pass diag_ops to the returned class, I didn't add that in the changes... Just do this one manually